src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -0,0 +1,36 @@
+package edu.kis.vh.nursery;
+
+public class DefaultCountingOutRhymer {
+
+	private static final int TABLE_SIZE = 12;
+	private static final int TOTAL_NUMBER = -1;
+	private static final int FULL_NUMBER = 11;
+	private final int[] numbers = new int[TABLE_SIZE];
+	private int total = TOTAL_NUMBER;
+
+	public void countIn(int in) {
+		if (!isFull())
+			numbers[++total] = in;
+	}
+
+	public boolean callCheck() {
+		return total == TOTAL_NUMBER;
+	}
+
+	public boolean isFull() {
+		return total == FULL_NUMBER;
+	}
+
+	protected int peekaboo() {
+		if (callCheck())
+			return TOTAL_NUMBER;
+		return numbers[total];
+	}
+
+	public int countOut() {
+		if (callCheck())
+			return TOTAL_NUMBER;
```

Comment:
Variable returned by peekaboo() and callCheck() is not the same number (even if the value is the same).

Reply:
Done